libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+            assert_response(response_dict=ip_result_dict, status_code=200)
+
+            networks = None
+
+            if ip_result_dict["metadata"]["network"] is not None:
+                networks = ip_result_dict["metadata"]["network"]["eth0"]
+
+                # the list of addresses
+                addresses = networks["addresses"]
+
+                for item in addresses:
+                    ips.append(item["address"])
+
+        metadata.update({"ips": ips})
+        return self._to_container(metadata=metadata)
+
+    def start_container(self, container, timeout=default_time_out):
+        """
+        Start a container
+
+        :param container: The container to start
+        :type  container: :class:`libcloud.container.base.Container`
+
+        :param timeout: Time to wait for the operation to complete
+        :type  timeout: ``int``
+
+        :rtype: :class:`libcloud.container.base.Container`
+        """
+        return self._do_container_action(container=container, action='start',
+                                         timeout=timeout,
+                                         force=True, stateful=True)
+
+    def stop_container(self, container, timeout=default_time_out):
+        """
+        Stop the given container
+
+        :param container: The container to be stopped
+        :type  container: :class:`libcloud.container.base.Container`
+
+        :param timeout: Time to wait for the operation to complete
+        :type  timeout: ``int``
+
+        :return: The container refreshed with current data
+        :rtype: :class:`libcloud.container.base.Container
+        """
+        return self._do_container_action(container=container, action='stop',
+                                         timeout=timeout,
+                                         force=True, stateful=True)
+
+    def restart_container(self, container, timeout=default_time_out):
+        """
+        Restart a deployed container
+
+        :param container: The container to restart
+        :type  container: :class:`.Container`
+
+        :param timeout: Time to wait for the operation to complete
+        :type  timeout: ``int``
+
+        :rtype: :class:`.Container`
+        """
+        return self._do_container_action(container=container, action='restart',
+                                         timeout=timeout,
+                                         force=True, stateful=True)
+
+    def destroy_container(self, container, timeout=default_time_out):
+        """
+        Destroy a deployed container. Raises and exception
+        if he container is running
+
+        :param container: The container to destroy
+        :type  container: :class:`.Container`
+
+        :param timeout: Time to wait for the operation to complete
+        :type timeout ``int``
+
+        :rtype: :class:`.Container`
+        """
+
+        # if the container is running then we cannot delete
+
+        if container.state == ContainerState.RUNNING:
```

Comment:
I'm fine with doing that check here although we don't really have clear rules / guidelines on how to handle such scenarios in Libcloud drivers.

A lot of drivers simply let the API handle that validation instead of doing it in the driver. This way it still works if user doesn't have the latest reference to the ``container`` object locally.

So another option would be to simply remove that check here.

Reply:
ok I can remove it then...